suse_migration_services/suse_product.py

```diff
@@ -117,3 +117,27 @@ def get_tag(self, tag):
                 'Parsing XML file {0} failed with: {1}'
                 .format(self.base_product, issue)
             )
+
+    def is_version_supported(self):
+        """Check that the product version to migrate is supported."""
+        base_patch = self.get_tag('version')[0].split(".")
+        is_supported = False
+        baseversion = int(base_patch[0])
+        if len(base_patch) == 2:
+            patch_level = int(base_patch[1])
+            min_base_supported = self.get_minimum_baseproduct_supported()
+            min_patch_supported = self.get_minimum_patch_level_supported()
+            if baseversion == min_base_supported and \
+               patch_level >= min_patch_supported:
+                is_supported = True
+
+        return is_supported
+
+    def get_default_target_version(self):
+        return 15
+
+    def get_minimum_baseproduct_supported(self):
+        return 12
+
+    def get_minimum_patch_level_supported(self):
+        return 3
```

Comment:
I do not understand the reason why this code is needed ? I think we limit ourselves with a constraint that is not needed. If a migration exists or not is defined by the image we deliver. There is no need to check on a minimum product version. Always think about this has to be maintained, I would just drop it

Reply:
Well, yes we had this discussion in some other  thread about SCC and the defined migration paths. My question is do we really want to continue to answer all the support questions? We've had a fair bit of churn already. Making the blast radius larger is not necessarily helpful. While the product QA team is happy to help us testing I don't think it is reasonable to test every release of the migration code with every origin point (SLES 12 SLES 12 SP1|2|3|4|5) to every destination (SLES 15 SLES SP1|2|3|4|5|6) the combination of which makes 42 test cases eventually. Things get worse when we consider there will be a next SLES.

I know, I know, the answer is that if it doesn't work it's probably the SCC team's problem. Problem with that approach is that we still end up answering all the questions. I think the list traffic proofs my point on this one.

I like a defined origin and destination. If the user over-writes the target in the config then we can waive our hands.

Yes, when $NEXT-SLES gets released we need to re-work the definition of origin and destination.

In short I like the definition of the parameters.

Reply:
> This is an important point and I would say yes. Any autodetection can fail. if it fails in the current implementation there is no way to fix this other than releasing a new image. So yes please treat an eventual given product definition from the custom config file as effective.
> 
From what I understood, I treated it as auto detection first and if failure, check custom product definition from user.
Not sure that is the preferred choice, where user could think something is wrong when their setting is ignored if auto detection worked.
IMO, the checking for a custom product should have priority and if no custom product, then auto detect the target.
Thoughts?

Reply:
> IMO, the checking for a custom product should have priority and if no custom product, then auto detect the target.

yep

Reply:
> Cloud Module enabled can install the migration packages,

I think you are wrong here. We had long discussions with the maintenance team when we recently had to release the system. There is a switch that keeps a package internal or not. suse-migration-services is a package that gets never released. It will not be available to customers on the channel. It only exists in the internal data structure (the pool) where we can access it. customers should not have the ability to "zypper in suse-migration-services"